src/components/LayerSwitcher/AddCustomLayer.tsx

```diff
@@ -218,6 +218,28 @@ const Details: React.FC<Detailsprops> = ({ layer, onChange, onValidation }) => {
   );
 };
 
+const humanJoin = (arr: string[]) => {
+  if (arr.length === 0) return '';
+  if (arr.length === 1) return arr[0];
+  if (arr.length === 2) return arr.join(' and ');
+
+  return arr.slice(0, -1).join(', ') + ' and ' + arr[arr.length - 1];
+};
+
+const MissingParts = ({ x, y, z }) => {
```

Comment:
Can we forward just the `url` prop here? It feels like it is responsibility of this component to check what is inside.